removeChildをしたとき、親のリストの長さをチェックするテストを作成

mukko opened the pull request.

mukko committed:
testRemoveChildメソッドを作成 (+1 squashed commit)
Squashed commits:
[be851b3] removeChildをしたとき、親のリストの長さをチェックするテストを作成

mukko committed:
removeChildをしたとき、親のリストの長さをチェックするテストを作成

sipo commented on test/GearStructureTest.hx:
```diff
@@ -67,4 +67,28 @@ class GearStructureTest
+	* removeChild後は0のはずである
+	**/
+
+	@Test("最初にremoveChildしたとき、親GearのchildGearListの長さが0である")
```
「最初にremoveChildしたとき」というのは条件の設定として実装と反している。「addChild後、removeChildすると」など

mukko committed:
testRemoveChildメソッドのアノテーションメッセージを修正

mukko committed:
GearStructureTestの共通操作であるparentとchildの生成、initializeTopをsetupに変更し、それに合わせプログラムを修正 (+3 squashed commits)
Squashed commits:
[bb5ed67] setupにinitializeTopの処理を追加

各テストメソッド内のsetup部分を削除
testInitializeTopにTODOを追加
[220fbf0] setupメソッドをテストするtestSetupメソッドを定義

不要なテストがあるかもしれない
[b47390e] [WIP]setupをオーバーライドした

setupのテストを作成する、テストメソッドを修正する

mukko committed:
[WIP]setupをオーバーライドした

setupのテストを作成する、テストメソッドを修正する

mukko committed:
setupメソッドをテストするtestSetupメソッドを定義

不要なテストがあるかもしれない

mukko committed:
setupにinitializeTopの処理を追加

各テストメソッド内のsetup部分を削除
testInitializeTopにTODOを追加

mukko committed:
testInitializeTopを修正、テストメソッド名も変更

mukko committed:
Merge branch 'testRemoveChild' of github.com:mukko/gipo into testRemoveChild
